### PR TITLE
143-feat: pre school translation

### LINKED ISSUE #143: Translation Pre-school course page
Translate Pre-school course page into the language of the course.
Maybe refactor 'courses.data.ts' file.
image

### PR DESCRIPTION
**Pull request by NikitaStarmoussov**
## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description

Updated page content

## Related Tickets & Documents

- Related Issue #143
- Closes #143

## Screenshots, Recordings

_Please replace this line with any relevant images for UI changes._

## Added/updated tests?

- [ ] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

### CODE REVIEW

File: src/features/about/about.data.ts [hunk 1]
```diff
@@ -118,20 +118,21 @@ const javaScriptPreSchoolRU: () => AboutInfo[] = () => {
...
-      title: 'Наставники и тренеры со всего мира',
-      info: 'Наставники и тренеры нашей школы - это фронтенд и разработчики JavaScript из разных компаний и стран. Как стать наставником?',
+      title: 'Время обучения',
+      info: 'Длительность обучения: 18 недель. Формат обучения: самообучение, групповое обучение, общение в Discord, задания проверяют в процессе кросс-чек и автоматически.',
       icon: planetIcon,
     },
     {
       id: 3,
       title: 'Бесплатное образование',
+      //TODO add correct info
```
Comment (dzmitry-varabei) [inline src/features/about/about.data.ts:128]: Правильный текст:
----
В RS School работает принцип "Pay it forward". Мы бесплатно делимся с учащимися своими знаниями сейчас, надеясь, что в будущем они вернутся к нам в качестве менторов и точно так же передадут свои знания следующему поколению студентов.

File: src/features/about/about.tsx [hunk 2]
```diff
@@ -52,6 +52,18 @@ export const About = ({ courseName, lang = 'en' }: AboutProps) => {
     <section className="course-about container">
       <div className="course-about content">
         <Title text={localizedContent[lang].title} />
+        {courseName === 'js / front-end pre-school ru' && (
```
Comment (dzmitry-varabei) [inline src/features/about/about.tsx:55]: looks like a hack )

File: src/features/certification/certification.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { Paragraph, Title } from '@/app/components';
+import { CourseNames } from '@/data/communication.data';
+
+import './certification.scss';
```
Comment (Quiddlee) [inline src/features/certification/certification.tsx:4]: Please, use css modules for styles - https://t.me/c/2142251410/1519

File: src/features/certification/certification.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  },
+};
+
+export const Certification = ({ courseName, lang = 'en' }: RequiredProps) => {
+  return (
+    <section className="certification container">
+      <article className="certification content info-wrapper">
+        <Title text={localizedContent[lang].title} hasAsterisk />
+        <Paragraph>{localizedContent[lang].firstParagraph}</Paragraph>
+        {courseName === 'js / front-end pre-school ru' && (
```
Comment (Quiddlee) [inline src/features/certification/certification.tsx:33]: Same for this line, should be used with constant

File: src/features/faq/faq.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import { TextWithLink, Title } from '@/app/components';
+
+import './faq.scss';
```
Comment (Quiddlee) [inline src/features/faq/faq.tsx:3]: Use css modules

File: src/features/faq/faq.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        title: 'чате',
+        link: 'https://discord.gg/2Ww3TCBvz4',
+      },
+    ],
+  },
+  {
+    question: 'Имеет ли значения город проживания? Можно ли пройти курс полностью онлайн?',
+    answer: 'Город проживания значения не имеет. Все этапы обучения можно пройти онлайн.',
+  },
+  {
```
Comment (Quiddlee) [inline src/features/faq/faq.tsx:32]: @ansivgit Do we have a conclusion about where the data should be stored, probably this shouldn't be inside the component file, but stored with other data in related place

File: src/features/required/required.tsx [hunk 7]
```diff
@@ -8,34 +8,25 @@ interface RequiredProps {
...
-          </div>
+          {knowBefore && 'description' in knowBefore && knowBefore.description.length !== 0 && (
+            <div>
+              <Subtitle text={knowBefore.title} />
+              <Actions actions={knowBefore.description} marked={marked1} />
+            </div>
+          )}
           {willLearn && 'description' in willLearn && willLearn.description.length !== 0 && (
             <div>
               <Subtitle text={willLearn.title} />
```
Comment (Quiddlee) [inline src/features/required/required.tsx:32]: My suggestion

```tsx
export const Required = ({ courseName, marked1, marked2 }: RequiredProps) => {
  const { knowBefore, willLearn, title } = courseDataMap[courseName];

  const isKnowBeforeExist =
    knowBefore && 'description' in knowBefore && knowBefore.description.length !== 0;

  const isWillLearnExist =
    willLearn && 'description' in willLearn && willLearn.description.length !== 0;

  return (
    <section className="required container">
      <div className="required content info-wrapper">
        <Title text={title} hasAsterisk />

        <div className="column-2">
          {isKnowBeforeExist && (
            <article>
              <Subtitle text={knowBefore.title} />
              <Actions actions={knowBefore.description} marked={marked1} />
            </article>
          )}
          {isWillLearnExist && (
            <article>
              <Subtitle text={willLearn.title} />
              <Actions actions={willLearn.description} marked={marked2} />
            </article>
          )}
        </div>
      </div>
    </section>
  );
};
```

File: src/pages/javascript-preschool-ru.tsx
Comment (Quiddlee) [inline src/pages/javascript-preschool-ru.tsx:1]: Replace this after pr [222](https://github.com/rolling-scopes/site/pull/222) will be merged, or you can merge this pr to your branch as well

image

https://github.com/rolling-scopes/site/pull/222/files#diff-7e3168e513d61f1758c26389577ff06ec9280381707c621730ead4a34f9329ae

File: src/pages/javascript-preschool-ru.tsx [hunk 8]
```diff
@@ -13,18 +13,17 @@ const COURSE_NAME = 'js / front-end pre-school ru';
...
   //todo trainers to ru
   return (
     <>
       <CourseMain courseName={COURSE_NAME} type={'Pre-school RU'} lang={lang} />
       <Breadcrumbs />
-      <TrainingProgram courseName={COURSE_NAME} lang={lang} />
       <About courseName={COURSE_NAME} lang={lang} />
+      <TrainingProgram courseName={COURSE_NAME} lang={lang} />
+      <Required courseName={COURSE_NAME} />
+      <Certification courseName={COURSE_NAME} lang="ru" />
```
Comment (Quiddlee) [inline src/pages/javascript-preschool-ru.tsx:24]:
```suggestion
      <Certification courseName={COURSE_NAME} lang={lang} />
```

File: src/pages/javascript-ru.tsx
Comment (Quiddlee) [inline src/pages/javascript-ru.tsx:1]: 🔄️🔄️🔄️

image

File: src/features/faq/faq.data.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  {
+    question: 'Обязательно ли смотреть вебинары школы?',
+    answer:
+      'Нет. Ссылки на рекомендуемую для изучения теорию находится в модулях. После самостоятельного изучения материалов модуля вы можете посмотреть вебинар, чтобы закрепить информацию или задать вопросы тренеру.',
+  },
+  {
+    question: 'Кто проверяет задания?',
+    answer:
+      'Проекты "CV", "Portfolio", "CodeJam", "JS 30" проверяются в ходе Cross-check. Алгоритмические таски и задачи из codewars проверяются автоматически. Ваши решения необходимо сабмитнуть в RS APP до дедлайна. Таски "CV#1. Markdown & Git", "CV#2. HTML, CSS & Git Basics" проверяются автоматически.',
```
Comment (natanchik) [inline src/features/faq/faq.data.ts:58]:
```suggestion
      'Проекты "CV", "Portfolio", "CodeJam", "JS 30" проверяются в ходе Cross-check. Алгоритмические задания, задачи из Codewars, а также задания "CV#1. Markdown & Git", "CV#2. HTML, CSS & Git Basics" проверяются автоматически. Ваши решения необходимо отправить в RS APP до дедлайна.',
```

Comment (natanchik) [inline src/features/faq/faq.data.ts:58]: I suggest choosing 1 designation option for `Cross-check` and using it throughout the site.

File: src/features/about/about.data.ts [hunk 10]
```diff
@@ -118,20 +117,20 @@ const javaScriptPreSchoolRU: () => AboutInfo[] = () => {
     },
     {
       id: 2,
-      title: 'Наставники и тренеры со всего мира',
-      info: 'Наставники и тренеры нашей школы - это фронтенд и разработчики JavaScript из разных компаний и стран. Как стать наставником?',
-      icon: planetIcon,
+      title: 'Время обучения',
+      info: 'Длительность обучения: 18 недель. Формат обучения: самообучение, групповое обучение, общение в Discord, задания проверяют в процессе кросс-чек и автоматически.',
```
[the remark this answers is not in the document]
Comment (NikitaStarmoussov) [reply src/features/about/about.data.ts:121]: Проверяются в процессе кросс-чек проверок?

File: src/features/faq/faq.data.ts [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  {
+    question: 'Можно ли пропускать вебинары?',
+    answer:
+      'Да, можно. Записи вебинаров можно будет найти на нашем канале - YouTube. Видео удобнее смотреть на скорости 1.25 или выше.',
+  },
+  {
+    question: 'Обязательно ли смотреть вебинары школы?',
+    answer:
+      'Нет. Ссылки на рекомендуемую для изучения теорию находится в модулях. После самостоятельного изучения материалов модуля вы можете посмотреть вебинар, чтобы закрепить информацию или задать вопросы тренеру.',
```
[the remark this answers is not in the document]
Comment (NikitaStarmoussov) [reply src/features/faq/faq.data.ts:53]: There were weekly streams where questions were answered.

File: src/features/about/about.data.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (natanchik) [reply src/features/about/about.data.ts:121]: ну да, тоже как-то не очень((
Ааа, 'check' это же 'проверка' 😂
Может `по средством Cross-check` или `в процессе Cross-check`? или `в процессе перекрестной проверки`, ну или оставить, но в остальных поменять, чтобы одинаково было

File: src/features/faq/faq.data.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (natanchik) [reply src/features/faq/faq.data.ts:53]: Maybe `куратору`?

File: src/features/about/about.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (NikitaStarmoussov) [reply src/features/about/about.tsx]: This is a lot of work that needs to be done as a separate task. This does not apply to the context of page translation.

File: src/features/faq/faq.module.scss [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+    font-size: 20px;
+    line-height: 28px;
+    color: $color-gray-600;
+    letter-spacing: 0;
+
+    .list {
+      @include media-tablet {
+        font-size: 18px;
+        font-weight: 400;
```
Comment (natanchik) [inline src/features/faq/faq.module.scss:19]:
```suggestion
        font-weight: $font-weight-regular;
```

File: src/features/required/required.tsx [hunk 13]
```diff
@@ -8,42 +8,57 @@ interface RequiredProps {
...
 };
+
+// (
+// willLearn.map((willLearn) => {
+// // <div>
+// //   <Subtitle text={willLearn.title} />
+// //   <Actions actions={willLearn.description} marked={marked2} />
+// // </div>
+// })
+// )
```
Comment (natanchik) [inline src/features/required/required.tsx:64]: could be removed

File: src/features/faq/faq.tsx (hunk 6 above)
Comment (ansivgit) [reply src/features/faq/faq.tsx]: I think it will be better to store it separate

**Review by ansivgit — APPROVED** [review]

**Review by natanchik — APPROVED** [review]

**Comment by ansivgit** [comment]
Please, fix PR name

**Review by Quiddlee — APPROVED** [review]

**Pull request merged**